ProcessMaker/processmaker
# Integrate Gmail SMTP Email Server

## Issue & Reproduction Steps

Ticket: [FOUR-6104](https://processmaker.atlassian.net/browse/FOUR-6104)

As a System Administrator, I want to enable an email server using the Gmail SMTP so that my platform utilizes my preferred and familiar email service

This PR updates the Settings UI based on the [Email Server PRD](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/2761064449/Email+Server+Options+-+Office365+Modern+Auth+and+Gmail+API) specifications as well as adds a new SMTP Authentication Method option allowing users to select either Standard Authentication or Google Authentication.

## Solution

- Remove Pagination
- Update the placement and verbiage of tab buttons
- Update 'Action' icons
- Adds new 'SMTP Authentication Method' UI config
- Adds the 'google' option in the UI for the 'SMTP Authentication Method' Model
- Implements XOAUTH2 for Gmail SMTP
- Adds migrations: `update_ui_in_settings_table` and `add_gmail_smtp_in_settings_table`

## UI Updates

**Before**

![Screen Shot 2022-07-14 at 11 07 28 AM](https://user-images.githubusercontent.com/52755494/179053935-eff9820a-472f-45e7-b4a6-a206f9d77759.png)
![Screen Shot 2022-07-14 at 11 07 39 AM](https://user-images.githubusercontent.com/52755494/179053975-d779a316-99ea-4af3-b0ce-61a02b326c9d.png)

**After**

![Screen Shot 2022-08-02 at 12 41 43 PM](https://user-images.githubusercontent.com/52755494/182459805-b612bddc-fb63-4dbe-9c5a-ca232f765149.png)

![Screen Shot 2022-08-02 at 12 42 35 PM](https://user-images.githubusercontent.com/52755494/182459893-66c6d73e-2277-4ccc-95a9-875b535c7c43.png)


## How to Test


**Testing UI and Migration**

1. Review the [PRD](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/2761064449/Email+Server+Options+-+Office365+Modern+Auth+and+Gmail+API)
2. Ensure you are on the `develop` branch for both Core and Send Email Connector
3. Ensure the following packages are installed - 
    - Auth Package
    - Send Email Connector [PR](https://github.com/ProcessMaker/connector-send-email/pull/209) `develop`
    - Actions By Email
    - DocuSign Connector
4. Configure a few Email servers with and without SMTP settings
5. Update Core branch `feature/FOUR-6104B` 
6. Update Send Email Connector branch `feature/FOUR-6104B`
7. Reinstall Send Email Connector
4. Run `php artisan migrate`
8. Within the settings tabs ensure UI displays properly as specified in the PRD
   - Buttons are displayed properly
   - Existing Email Servers Configurations display properly
   - Add / Remove Email Servers works as expected
   - Configurations Update as expected. etc...


## Related Tickets & Packages
- Ticket: [FOUR-6104](https://processmaker.atlassian.net/browse/FOUR-6104)
- [Email Server PRD](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/2761064449/Email+Server+Options+-+Office365+Modern+Auth+and+Gmail+API)
- ~~[Send Email PR](https://github.com/ProcessMaker/connector-send-email/pull/209)~~ `feature/FOUR-6104`
- [Send Email NEW PR](https://github.com/ProcessMaker/connector-send-email/pull/211) `feature/FOUR-6104B`

ci:connetor-send-email:feature/FOUR-6104B

## Code Review Checklist
